lib/ansible/playbook/role/__init__.py

```diff
@@ -526,15 +537,14 @@ def get_all_dependencies(self):
         Returns a list of all deps, built recursively from all child dependencies,
         in the proper order in which they should be executed or evaluated.
         '''
+        if not self._all_dependencies:
```

Comment:
Should we rather set `self._all_dependencies` to `None` initially and check here for `is None`? That way if there are no `self._all_depedencies` (`== []`) the `for` loops wouldn't even get attempted to execute. Probably doesn't matter with the current code but thought I'd mention it.